src/js/utils/time-ranges.js

```diff
@@ -25,8 +35,22 @@ export function createTimeRange(start, end){
     };
   }
   return {
-    length: 1,
-    start: function() { return start; },
-    end: function() { return end; }
+    length: ranges.length,
+    start: getRange.bind(null, 'start', 0, ranges),
+    end: getRange.bind(null, 'end', 1, ranges)
   };
 }
+
+function getRange(fnName, valueIndex, ranges, rangeIndex){
+  if (rangeIndex === undefined) {
```

Comment:
@heff I added the deprecation warning, none of our plugins use the no-index style but we can't be sure that no one else has. I'd rather we provide a clear indication that we intend to remove that "feature" in the near future.